### PR TITLE
Recipient can register for an account

### PR DESCRIPTION
**Pull request by FlisAnn**
[MOBILE: Recipient can register for an account](https://www.pivotaltracker.com/story/show/176623674)
### User Story
```
As an visitor,
In order to access the application,
I would like the site to have a unique way for a visitor to register
```
## Changes proposed in this pull request:
* Adding Registration Form 
* Adding functionality to register for an account

## What I have learned working on this feature:
* How to register for an acount in react native

## Screenshots (Client)
Screenshot 2021-02-02 at 13 14 01

### CODE REVIEW

File: components/DisplayFoodBag.jsx [hunk 1]
```diff
@@ -2,8 +2,8 @@ import React from 'react'
 import { StyleSheet, View, Text, Dimensions } from 'react-native'
 import { useSelector } from 'react-redux'
 
-const DisplayFoodBagList = ({ foodbag }) => {
-  const { credentials } = useSelector((state) => state)
+const DisplayFoodBag = ({ foodbag }) => {
+  const { credentials } = useSelector(state => state)
```
Comment (psnoya) [inline components/DisplayFoodBag.jsx:6]: fomatting

File: modules/auth.js [hunk 2]
```diff
@@ -1,8 +1,9 @@
 import axios from 'axios'
-import store from '../state/store/store'
 import AsyncStorage from '@react-native-community/async-storage'
+const apiUrl =
+  process.env.NODE_ENV === 'development' &&
+  'https://dream-time-news-api.herokuapp.com/api0'
```
Comment (swardbladpaulo) [inline modules/auth.js:5]: Is this the right api path to heroku???

Comment (psnoya) [inline modules/auth.js:5]: ???

File: modules/auth.js [hunk 3]
```diff
@@ -35,8 +36,8 @@ class Auth {
         ? this.options.authUrl.validateToken
         : '/validate_token'
     }`
-    axios.interceptors.response.user(
-      (response) => {
+    axios.interceptors.response.use(
```
Comment (psnoya) [inline modules/auth.js:39]: correct use to user

File: modules/auth.js [hunk 4]
```diff
@@ -68,6 +69,23 @@ class Auth {
...
+    return new Promise(async (resolve, reject) => {
+      try {
+        const signUpResponse = await axios.post(
+          this.apiAuthUrl,
+          {
+            ...userFields,
+          },
+          { params: { confirm_success_url: confirmSuccessUrl } }
+        )
+        this.setSession(signUpResponse.headers)
```
Comment (psnoya) [inline modules/auth.js:82]: formatting

File: modules/auth.js [hunk 5]
```diff
@@ -96,17 +144,93 @@ class Auth {
...
+          url,
+          method: options.method,
+          data: options.data,
+          params: options.params,
+          headers: {
+            ...options.headers,
+            ...this.session,
+          },
+        })
+        this.setSession(reponse.headers)
```
Comment (psnoya) [inline modules/auth.js:224]: reponse typo = should be response

**Review by psnoya — COMMENTED** [review]
hippity hoppity

**Pull request merged**